### ISSUE TITLE
Error when creating mock of enum with negative value since Typescript 5.4

### ISSUE THREAD

**Issue by hoi4** [Typescript-TDD/ts-auto-mock#1579]
### Error when creating mock of enum with negative value since Typescript 5.4
When upgrading from Typescript `5.3.3` to `5.4.5`, some of our tests are failing with the following error message:

`Debug Failure. False expression: Negative numbers should be created in combination with createPrefixUnaryExpression`

This is caused by calling `createHydratedMock` with an interface that includes a property of an enum type with a negative value, e.g.

```typescript
enum TestEnum {
  VALUE_ONE = -1, // <-- This negative enum value seems to be the problem
  VALUE_TWO = 0,
  VALUE_THREE = 1
}

interface Test {
  readonly enumProp: TestEnum;
}

// in the test
const test = createHydratedMock<Test>(); // errors
```

### Your environment
* ts-auto-mock version: `3.7.2`
* typescript version: `5.4.5`
* node version: `v18.19.0`
* npm version: `10.2.3`

**Comment by uittorio** [comment]
Hey! Thank you for raising the issue! I will have look and let you know!

**Comment by uittorio** [comment]
Just released a new version to address the issue!! thank you again! 
Released in 3.7.3
https://github.com/Typescript-TDD/ts-auto-mock/blob/master/CHANGELOG.md

**Comment by hoi4** [comment]
Works like a charm! I greatly appreciate the quick fix! Thank you! 🚀
